atflow/dataset.py

```diff
@@ -124,6 +125,7 @@ def has_same_batch(data):
         self._minibatch_indicies = None
         self.info = kwargs
         self.next_epoch()
+        self.normalizer = lambda data: [d for d in data] if isinstance(data, list) else data
```

Comment:
Why not just `lambda data: data`? Given that it is likely a list of numpy array (if it's a list), the new list is still linked to the original list anyway. Also, this version cannot deal with more complex nested list.